frontend/src/metabase/admin/settings/components/widgets/VersionUpdateNotice/VersionUpdateNotice.tsx

```diff
@@ -0,0 +1,161 @@
+import cx from "classnames";
+import { c, t } from "ttag";
+
+import {
+  getCurrentVersion,
+  getLatestVersion,
+} from "metabase/admin/settings/selectors";
+import { useAdminSetting } from "metabase/api";
+import { useSetting } from "metabase/common/hooks";
+import ExternalLink from "metabase/core/components/ExternalLink";
+import ButtonsS from "metabase/css/components/buttons.module.css";
+import CS from "metabase/css/core/index.css";
+import { useSelector } from "metabase/lib/redux";
+import { newVersionAvailable, versionIsLatest } from "metabase/lib/utils";
+import type { VersionInfoRecord } from "metabase-types/api";
+
+import S from "./VersionUpdateNotice.module.css";
+
+export function VersionUpdateNotice() {
+  const currentVersion = useSelector(getCurrentVersion);
+  const latestVersion = useSelector(getLatestVersion);
+  const isHosted = useSetting("is-hosted?");
+  // This should not be here
+  useAdminSetting("site-url");
```

Comment:
For some reason, if I don't include this hook the tests in `VersionUpdateNotice/VersionUpdateNotice.tsx` don't work, because the mock settings don't initialise properly

Reply:
Is the problem that `settings` are loaded from `api/session/properties` and that can cause issues if a component expects `settings` to always be populated?

I don't think I understand your suggestions because they still seem to rely on the `settings` slice being loaded, e.g. `useSetting` is just a wrapper around `getSetting` and `useSelector`
```
 useSelector((state) => getSetting(state, settingName));
 ```